modules/core/src/main/java/org/apache/ignite/internal/managers/deployment/GridDeploymentClassLoader.java

```diff
@@ -615,8 +622,20 @@ else if (log.isDebugEnabled())
                     continue;
                 }
 
-                if (res.success())
+                if (res.success()) {
+                   
+                    // If we searched other nodes w/o succeeding first, then 
+                    // make the node we found the resource on the first to search next time.
```

Comment:
I would consider replacing "node" with "participant" in all the comments you've added as you did that above and it's a much clearer description than "node"